sign/cosi/cosi.go

```diff
@@ -307,6 +307,27 @@ func (m *Mask) SetBit(i int, enable bool) error {
 	return nil
 }
 
+// IndexEnabled checks whether the given index is enabled in the mask or not.
+func (m *Mask) IndexEnabled(i int) (bool, error) {
+	if i >= len(m.publics) {
+		return false, errors.New("index out of range")
+	}
+	byt := i >> 3
+	msk := byte(1) << uint(i&7)
```

Comment:
Sorry to chime in so late: what about:

```
byt := i / 8
msk := byte(1 << ( i % 8 ))
```

It's the same, but I like the division and modulo 8, it tells me better what's going on than right-shifting by 3 and and with 7...